src/sys/unix.rs

```diff
@@ -1479,6 +1501,20 @@ impl crate::Socket {
         }
     }
 
+    /// Sets `SO_PEERCRED` to null on the socket.
+    /// It disables the initial handshake of unix domain sockets.
```

Comment:
Can you please improve the documentation for this. It should explain that normally cygwin performs an `AF_LOCAL` handshake when opening local sockets, and that this option turns off that cygwin extension. Also please include a link to the discussion thread about it.

It would also make sense to document on `connect` about this behavior. It's unfortunate that cygwin differs like this, but if that's how the OS works, then I'm not sure we can do much about it.

Reply:
FYI, I have https://github.com/llvm/llvm-project/pull/140353 for the same issue.  It only seems to be a problem in the tests though (I think any real-world case would have at least separate threads if not processes involved, or they could use `socketpair` which takes care of getting both sides of the socket connected to each other at once)

Reply:
OK, docs are added for `set_no_peercred`, `connect` and `accept`.